core/common/src/format/DateTimeFormatBuilder.kt

```diff
@@ -129,6 +129,8 @@ public sealed interface DateTimeFormatBuilder {
          *
          * [am] is used for the AM marker (0-11 hours), [pm] is used for the PM marker (12-23 hours).
          *
+         * Empty strings can not be used as markers.
```

Comment:
Do we check this requirement in the `amPmMarker` call, or it fails the operation (the parsing I suppose) later?
Do we need the requirement to have distinct strings here?

Reply:
The `amPmMarker` call checks this requirement. Documented this and added the tests.
Also, added the requirement for distinct strings for month days, days of the week, and AM/PM.